src/CaptureFileInfo/include/CaptureFileInfo/CaptureFileInfo.h

```diff
@@ -22,16 +24,21 @@ class CaptureFileInfo {
 
   [[nodiscard]] QDateTime LastUsed() const { return last_used_; }
   [[nodiscard]] QDateTime Created() const { return file_info_.birthTime(); }
+  [[nodiscard]] absl::Duration CaptureLength() const { return capture_length_; }
 
   [[nodiscard]] bool FileExists() const;
 
   [[nodiscard]] uint64_t FileSize() const;
 
   void Touch() { last_used_ = QDateTime::currentDateTime(); }
+  void SetCaptureLength(absl::Duration capture_length) { capture_length_ = capture_length; }
+
+  static constexpr absl::Duration kMissingCaptureLengthValue{absl::Nanoseconds(0)};
 
  private:
   QFileInfo file_info_;
   QDateTime last_used_;
+  absl::Duration capture_length_ = kMissingCaptureLengthValue;
```

Comment:
Please use `std::optional<absl::Duration>` here. Using std::optional makes it easy to understand that this value can be unavailable. It also makes the rest of the code easier.